File: VERMEG-HIVE/src/main/java/com/vermeg/services/questions/QuestionService.java
package com.vermeg.services.questions;
import com.vermeg.entities.Questions;

import java.util.List;
import java.util.Optional;

public interface QuestionService {

    List<Questions> getAllQuestions();

    Optional<Questions> getQuestionById(Long id);

    List<Questions> searchQuestions(String keyword);

    Questions saveQuestion(Questions question);

    Questions updateQuestion(Long id, Questions updatedQuestion);

    void deleteQuestion(Long id);

    Questions upVoteQuestion(Long questionId);

    Questions downVoteQuestion(Long questionId);
}

File: vermeg-hive-angular/src/app/question/question.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Question } from './question.model';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {
  private apiUrl = 'http://localhost:8080/api/questions';

  constructor(private http: HttpClient) { }

  getAllQuestions(): Observable<Question[]> {
    return this.http.get<Question[]>(`${this.apiUrl}`);
  }

  getQuestionById(id: number): Observable<Question> {
    return this.http.get<Question>(`${this.apiUrl}/${id}`);
  }

  createQuestion(question: Question): Observable<Question> {
    return this.http.post<Question>(`${this.apiUrl}`, question);
  }

  upvoteQuestion(questionId: number) {
    return this.http.post<Question>(`${this.apiUrl}`, questionId);

  }

  downvoteQuestion(questionId: number) {
    return this.http.post<Question>(`${this.apiUrl}`, questionId);

  }
}
